syntax = "proto3";

service FullNode {
  rpc Handshake(HandshakeRequest) returns (HandshakeResponse);
  rpc NewTransactionBroadcast(NewTransactionRequest) returns (Empty);
  rpc NewBlockBroadcast(NewBlockRequest) returns (Empty);
}

message HandshakeRequest {
  string version = 1;
  string time = 2;
  string addrMe = 3;
  int32 bestHeight = 4;
}

message HandshakeResponse { repeated string knownPeers = 1; }

message NewTransactionRequest { string transaction_data = 1; }

message NewBlockRequest { string block_data = 1; }

message Empty {}
